### DISCUSSION TITLE
Chafa viewer triggers password lock icon and security indicator

### DISCUSSION THREAD

**Discussion by edrozenberg** [ghostty-org/ghostty#3980] [Feature Requests, Ideas] [answered]
When using the "chafa" viewer, it triggers Ghostty's lock and security indicator to appear. When viewing animated gifs this happens until the user quits chafa w/ CTRL-C. When viewing a static image this happens briefly until the image has been displayed.

Animated gif behavior:

https://github.com/user-attachments/assets/c7a41055-07b7-4142-9a06-23e2e340657b

This may be some kind of chafa issue not sure, but there is no password or security involved in viewing available items with chafa, so is Ghostty somehow mis-recognizing a security event/context by something chafa is doing.

No such issue with other CLI apps or viewers (no issue w/ "viu", or "yazi" etc).

**Comment by mitchellh** [comment, marked as answer]
This is based on the heuristic we use to detect password input. I don't think there's a way for us to fix this because that's the same heuristic used by iTerm and others to enable this feature. I would recommend reporting to Chafa to try to ease up on their ioctls here.

**Reply by edrozenberg** [reply]
Thank you, opened a discussion at Chafa on this (# 236).

**Reply by edrozenberg** [reply]
Chafa will have a fix for this. From Chafa:
>[pjansson] Maintainer
> ... When tcsetattr() is used to turn off echo, ghostty interprets it to mean there's a password prompt. You can indeed work around it by passing --polite on. I'll change it to emit a control sequence instead, which does not trigger the heuristic.

**Reply by edrozenberg** [reply]
The initial chafa fix was not correct. It is now fixed properly by chafa in chafa commit 59409a9:
> chafa: Proper workaround for ghostty emoji on echo off
> Disable canonical mode along with echo. This works around the ghostty
heuristic that causes it to emit emoji when echo is turned off.
